### PR TITLE
Generalize and Add Support for Bundling

### PR DESCRIPTION
**Pull request by trek**
* Removes ember-specific defaults
* Adds initial support for bundling

### CODE REVIEW

**Comment by ef4** [comment]
This looks nice.

**Comment by trek** [comment]
This is ready for review. I'd like to add support for concat in another PR to keep this one small (it's already p big)

File: README.md [hunk 1]
```diff
@@ -1,21 +1,135 @@
...
+So, if you have the following tree:
+
+```
+
+├── inner
+│   └── first.js
+└── outer.js
+```
+
+You will have the the following module names passed to AMD's `define` call:
```
Comment (chadhietala) [inline README.md]: Remove extra "the"

**Comment by rwjblue** [comment]
Released as 0.5.0.  Updating ember-cli now.

**Comment by rwjblue** [comment]
https://github.com/ember-cli/ember-cli/pull/3252

**Pull request merged**